x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Verifier.java

```diff
@@ -626,6 +628,15 @@ private static boolean checkGroupMatch(Expression e, Node<?> source, List<Expres
         return false;
     }
 
+    private static void checkBooleanFiltering(LogicalPlan p, Set<Failure> localFailures) {
+        if (p instanceof Filter) {
+            Expression condition = ((Filter) p).condition();
+            if (condition.resolved() && condition.dataType() != BOOLEAN) {
```

Comment:
No need to check if the condition is resolved, this has already been checked before - these checks get applied only if the plan is fully resolved.

Reply:
Right, this follows after checking against unresolved nodes (and even a comment about it!). Thanks, fixed.